### ISSUE TITLE
flutter build ios-framework failed while use shared_preferences: 0.5.6+1

### ISSUE THREAD

**Issue by Lishaoshuai** [flutter/flutter#51042]
## Steps to Reproduce

1. use shared_preferences, version: 0.5.6+1
2. flutter build ios-framework

## Logs

```
FileSystemException: FileSystemException: Directory listing failed, path = 'my_product_path/output/Debug/iphoneos/Debug-iphoneos/' (OS Error: No such file or directory, errno = 2)
```

```
#0      _Directory._fillWithDirectoryListing (dart:io-patch/directory_patch.dart:39:68)
#1      _Directory.listSync (dart:io/directory_impl.dart:251:5)
#2      ForwardingDirectory.listSync (package:file/src/forwarding/forwarding_directory.dart:45:12)
#3      BuildIOSFrameworkCommand._producePlugins (package:flutter_tools/src/commands/build_ios_framework.dart:390:61)
<asynchronous suspension>
#4      BuildIOSFrameworkCommand.runCommand (package:flutter_tools/src/commands/build_ios_framework.dart:175:15)
<asynchronous suspension>
#5      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:615:18)
<asynchronous suspension>
#6      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:515:33)
<asynchronous suspension>
#7      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:146:29)
#8      _rootRun (dart:async/zone.dart:1126:13)
#9      _CustomZone.run (dart:async/zone.dart:1023:19)
#10     _runZoned (dart:async/zone.dart:1518:10)
#11     runZoned (dart:async/zone.dart:1465:12)
#12     AppContext.run (package:flutter_tools/src/base/context.dart:145:18)
#13     FlutterCommand.run (package:flutter_tools/src/runner/flutter_command.dart:505:20)
#14     CommandRunner.runCommand (package:args/command_runner.dart:197:27)
#15     FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:413:21)
<asynchronous suspension>
#16     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:146:29)
#17     _rootRun (dart:async/zone.dart:1126:13)
#18     _CustomZone.run (dart:async/zone.dart:1023:19)
#19     _runZoned (dart:async/zone.dart:1518:10)
#20     runZoned (dart:async/zone.dart:1465:12)
#21     AppContext.run (package:flutter_tools/src/base/context.dart:145:18)
#22     FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:364:19)
#23     CommandRunner.run.<anonymous closure> (package:args/command_runner.dart:112:25)
#24     new Future.sync (dart:async/future.dart:224:31)
#25     CommandRunner.run (package:args/command_runner.dart:112:14)
#26     FlutterCommandRunner.run (package:flutter_tools/src/runner/flutter_command_runner.dart:250:18)
#27     run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:63:22)
#28     _rootRun (dart:async/zone.dart:1126:13)
#29     _CustomZone.run (dart:async/zone.dart:1023:19)
#30     _runZoned (dart:async/zone.dart:1518:10)
#31     runZoned (dart:async/zone.dart:1502:12)
#32     run.<anonymous closure> (package:flutter_tools/runner.dart:61:18)
<asynchronous suspension>
#33     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:146:29)
#34     _rootRun (dart:async/zone.dart:1126:13)
#35     _CustomZone.run (dart:async/zone.dart:1023:19)
#36     _runZoned (dart:async/zone.dart:1518:10)
#37     runZoned (dart:async/zone.dart:1465:12)
#38     AppContext.run (package:flutter_tools/src/base/context.dart:145:18)
#39     runInContext (package:flutter_tools/src/context_runner.dart:64:24)
#40     run (package:flutter_tools/runner.dart:50:10)
#41     main (package:flutter_tools/executable.dart:65:9)
#42     main (file:///Users/shaoshuaili/Desktop/flutter/packages/flutter_tools/bin/flutter_tools.dart:8:3)
#43     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:305:32)
#44     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:174:12)
```
while i add --verbose

```
[   +1 ms] executing: [my_product_path/.ios/Pods/] xcrun xcodebuild
-alltargets -sdk iphonesimulator -configuration Debug
SYMROOT=my_product_path/output/Debug/iphonesimulator ARCHS=x86_64
ONLY_ACTIVE_ARCH=NO
[+1302 ms] Build settings from command line:
                        ARCHS = x86_64
                        ONLY_ACTIVE_ARCH = NO
                        SDKROOT = iphonesimulator13.0
                        SYMROOT =
                        /output/Debug/iphonesimulator

                    note: Using new build system


                    2020-02-19 22:27:41.114 xcodebuild[19047:11291885] unable to create build operation: Error
                    Domain=com.apple.xcbuild Code=1 "unknown error while handling message: missingTarget(guid:
                    "aaed406f79e3587e78254d18b1de058558e43d378bdedc3ceb60c1adb0f770ac")"
                    UserInfo={NSLocalizedDescription=unknown error while handling message: missingTarget(guid:
                    "aaed406f79e3587e78254d18b1de058558e43d378bdedc3ceb60c1adb0f770ac")}
                    ** BUILD FAILED **
```

```
[32m[✓][39m Flutter (Channel stable, v1.12.13+hotfix.5, on Mac OS X 10.15.2 19C57, locale zh-Hans-CN)
    [32m•[39m Flutter version 1.12.13+hotfix.5 at /Users/shaoshuaili/Desktop/flutter
    [32m•[39m Framework revision 27321ebbad (2 months ago), 2019-12-10 18:15:01 -0800
    [32m•[39m Engine revision 2994f7e1e6
    [32m•[39m Dart version 2.7.0

[33m[!][39m Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    [32m•[39m Android SDK at /Users/shaoshuaili/Library/Android/sdk
    [32m•[39m Android NDK location not configured (optional; useful for native profiling support)
    [32m•[39m Platform android-29, build-tools 29.0.2
    [32m•[39m Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    [32m•[39m Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    [33m![39m Some Android licenses not accepted.  To resolve this, run: flutter doctor --android-licenses

[32m[✓][39m Xcode - develop for iOS and macOS (Xcode 11.0)
    [32m•[39m Xcode at /Applications/Xcode11.app/Contents/Developer
    [32m•[39m Xcode 11.0, Build version 11A420a
    [32m•[39m CocoaPods version 1.8.4

[32m[✓][39m Android Studio (version 3.5)
    [32m•[39m Android Studio at /Applications/Android Studio.app/Contents
    [32m•[39m Flutter plugin version 38.2.3
    [32m•[39m Dart plugin version 191.8423
    [32m•[39m Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[32m[✓][39m VS Code (version 1.41.1)
    [32m•[39m VS Code at /Applications/Visual Studio Code.app/Contents
    [32m•[39m Flutter extension version 3.8.1

[33m[!][39m Proxy Configuration
    [32m•[39m HTTP_PROXY is set
    [33m![39m NO_PROXY is not set

[32m[✓][39m Connected device (1 available)
    [32m•[39m iPhone 11 • 9697FD3D-8CB1-40E3-9ED2-6E8755A48068 • ios • com.apple.CoreSimulator.SimRuntime.iOS-13-0 (simulator)

[33m![39m Doctor found issues in 2 categories.
```

**Comment by BaderEddineOuaich** [comment]
Try using the absolute path

**Comment by Lishaoshuai** [comment]
> Try using the absolute path

i use the absolute path. if remove shared_preferences, it worked.

**Comment by jmagman** [comment]
This is a duplicate of https://github.com/flutter/flutter/issues/46993 which was fixed in 1.13.3.  The workaround is to pass in an absolute path in `--output`

```
flutter build ios-framework --output=/absolute/path/to/build_ios_test
```

Closing.
